### PR TITLE
feat: Adding the NFBServer class

### LINKED ISSUE #194: Missing NFBServer
**Actual Behaviour**
Missing NFBServer. This class is one of the necessities and has instanced usage in most of the rendering classes.

**Expected Behaviour**
This should be present,

**Would you like to work on the issue?**
Yes, I work on it.

### PR DESCRIPTION
**Pull request by jddeep**
Fixes #194 partly.

**Changes**: Started porting and implementing the NFBServer class. Currently, it has functionality related to those which are going to be used in FileOutputTask class. 

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

### CODE REVIEW

File: app/src/main/java/io/neurolab/NFBServer.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    public static int samplesPerSecond = 256;
+    public static int numChannels = 2;
+
+    private List<double[]> inputData;
+    protected ConcurrentLinkedDeque<double[]> currentData = new ConcurrentLinkedDeque<double[]>();
+    private int numSamples;
+    private static ReentrantLock lock = new ReentrantLock();
+    public int minimumNewSamples = 16;
+    private LinkedList<Task> tasks;
+    private long lastTimestamp;
```
Comment (adityastic) [inline app/src/main/java/io/neurolab/NFBServer.java:26]: Where is being used?

Comment (jddeep) [reply app/src/main/java/io/neurolab/NFBServer.java:26]: Check out the append data method. Currently we are only initializing this there, but we will use it in the successive PRs directed to update this class once FileOutputTask get's successfully created without any issue. Removing this right now won't have much effect as we need to add it in the very near future again then. I have kept this to finalize the appendData method to the max extent.

Comment (adityastic) [reply app/src/main/java/io/neurolab/NFBServer.java:26]: @jddeep ok cool

**Review by adityastic — APPROVED** [review]

**Comment by mariobehling** [comment]
If you want these to be reviewed please take out "WIP".

**Comment by jddeep** [comment]
> If you want these to be reviewed please take out "WIP".

Alright @mariobehling Done.

**Review by suresh-gandhi — COMMENTED** [review]
@jddeep Changes are good. Try improvising the Codacy/PR quality review. I guess it has some unused variables.

**Review by mariobehling — APPROVED** [review]

**Pull request merged**